### ISSUE TITLE
LazyTensor.ndim not giving correct value

### ISSUE THREAD

**Issue by JonasDeSchouwer** [getkeops/keops#376]
The title says it all.

Minimal example:
```python
import torch
from pykeops.torch import LazyTensor

x = LazyTensor(torch.rand((5,1,8)))

print(x.shape)  # (5,1,8)   -> correct
print(x.ndim)   # 8         -> incorrect: always x.shape[-1]
```

As you can see, the last print should have '3' as output, but instead has '8'. Some experimenting showed that the returned value seems to always be `x.shape[-1]`.

My environment:
- Windows + WSL (Ubuntu 24.04)
- torch: 2.2.2+cu121
- pykeops: 2.2.3

**Comment by huguesva** [comment]
As a remark, I noticed the same thing and resorted to using `len(x.shape)`.

**Comment by JonasDeSchouwer** [comment]
Yes, I did the same. But it would of course be nicer (and not lead to weird errors for developers expecting the normal behaviour)
